addon/serializers/drf.js

```diff
@@ -11,6 +11,50 @@ import Ember from 'ember';
  * @extends DS.RESTSerializer
  */
 export default DS.RESTSerializer.extend({
+  /**
+   * Normalizes a part of the JSON payload returned by the server. This
+   * version simply calls addRelationshipsToLinks() before invoking
+   * the RESTSerializer's version.
+   *
+   * @method normalize
+   * @param {subclass of DS.Model} typeClass
+   * @param {Object} hash
+   * @param {String} prop
+   * @return {Object}
+   */
+  normalize: function(typeClass, hash, prop) {
+    this.addRelationshipsToLinks(typeClass, hash);
+    return this._super(typeClass, hash, prop);
+  },
+
+  /**
+   *  Adds relationships to the links hash as expected by the RESTSerializer.
+   *
+   * @method addRelationshipsToLinks
+   * @private
+   * @param {subclass of DS.Model} typeClass
+   * @param {Object} hash
+   */
+  addRelationshipsToLinks: function(typeClass, hash) {
+    if (!hash.hasOwnProperty('links')) {
+      hash['links'] = {};
+    }
+
+    typeClass.eachRelationship(function (key, relationship) {
+      let payloadRelKey = this.keyForRelationship(key);
+      if (!hash.hasOwnProperty(payloadRelKey)) {
+        return;
+      }
+      if (relationship.kind === 'hasMany' || relationship.kind === 'belongsTo') {
+        // Matches strings starting with: https://, http://, //, /
+        var payloadRel = hash[payloadRelKey];
+        if (!Ember.isNone(payloadRel.match) && typeof(payloadRel.match) === 'function' && payloadRel.match(/^((https?:)?\/\/|\/)\w/)) {
```

Comment:
I am getting `TypeError: Cannot read property 'match' of null` here.  I think it is from one of my records that has an empty belongsTo.  I set up a try/catch/debugger and captured this info:

```
relationship.kind = "belongsTo"
payloadRelKey = "myModel"
payloadRel = null
```

I think you just need to account for the null relationship.